actioncable/lib/action_cable/storage_adapter/base.rb

```diff
@@ -0,0 +1,22 @@
+module ActionCable
+  module StorageAdapter
+    class Base
+      attr_reader :logger, :server
+
+      def initialize(server)
+        @server = server
+        @logger = @server.logger
+      end
+
+      # Storage connection instance used for broadcasting. Not intended for direct user use.
+      def broadcast
+        raise NotImplementedError
+      end
+
+      # Storage connection instance used for pubsub.
+      def pubsub
+        raise NotImplementedError
+      end
```

Comment:
IMHO, you shouldn't be calling `publish`, `subscribe`, etc. directly on the adapter -- the adapter is simply the connection holder, not the means of communication with the backend directly.